src/nvim/tui/input.h

```diff
@@ -14,6 +15,8 @@ typedef struct term_input {
   bool waiting;
   bool ttimeout;
   int8_t waiting_for_bg_response;
+  int8_t waiting_for_extkeys_response;
+  bool supports_kitty_protocol;
```

Comment:
is kitty protocol different than extkeys? Or can we call it extkeys generically?

Reply:
This might be more information than you bargained for, but:

`modifyOtherKeys` is the name of the Xterm extension to disambiguate control sequence modifiers. This mode can be programmatically enabled via the XTMODKEYS sequence `\x1b[>4;2m`.

"CSI u" is the colloquial name given to [Leonerd's encoding scheme](http://www.leonerd.org.uk/hacks/fixterms/), which is distinct from Xterm's `modifyOtherKeys`. `libtermkey` (which Neovim uses) supports both.

Kitty's keyboard protocol is yet another encoding scheme which is based off of Leonerd's CSI u. Kitty's protocol supports progressive enhancement so you can use different "levels" of the protocol; the base level (level 1, which this PR enables) is identical to CSI u.

However, Leonerd's CSI u proposal does not provide any way of querying the terminal for support nor programmatically enabling or disabling that mode. Both of these are supported by kitty's keyboard protocol.

Finally, tmux uses the term "extended keys" to refer to any encoding scheme that disambiguates control sequences (either `modifyOtherKeys` or CSI u). I've adopted that terminology to refer to the same thing.

So here "extended keys" (extkeys) means *any* encoding scheme that enables us to distinguish between e.g. `<Tab>` and `<C-I>`. `modifyOtherKeys` is the Xterm scheme and kitty/CSI u is the one proposed by Leonerd.

The terminology is kind of confusing and there isn't a super well established standard. As I said I'm open to suggestions on wording, but for now I will use the terms as outlined above.

I'll also add a section to the docs outlining all of this.

Reply:
> However, Leonerd's CSI u proposal does not provide any way of querying the terminal for support nor programmatically enabling or disabling that mode. Both of these are supported by kitty's keyboard protocol.

Do we actually need that? Why not always try to enable it?
 
> Finally, tmux uses the term "extended keys" to refer to any encoding scheme that disambiguates control sequences (either `modifyOtherKeys` or CSI u). I've adopted that terminology to refer to the same thing.

👍  good choice

Reply:
> Do we actually need that? Why not always try to enable it?

Well we'd need to know if the terminal understood our attempt to enable it so that if it doesn't we can fall back to the `modifyOtherKeys` sequence. Since that would require querying the terminal anyway, we might as well just ask up front if it supports it.

Reply:
> Kitty's keyboard protocol is yet another encoding scheme which is based off of Leonerd's CSI u. Kitty's protocol supports progressive enhancement so you can use different "levels" of the protocol; the base level (level 1, which this PR enables) is identical to CSI u.
> 
> However, Leonerd's CSI u proposal does not provide any way of querying the terminal for support nor programmatically enabling or disabling that mode. Both of these are supported by kitty's keyboard protocol.

There are a whole bunch of problems with fixterms that are addressed by kitty's protocol, see https://sw.kovidgoyal.net/kitty/keyboard-protocol/#fixterms-bugs